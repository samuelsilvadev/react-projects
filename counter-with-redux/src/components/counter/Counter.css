.counter {
    display: flex;
    flex-wrap: wrap;
    margin-top: 50vh;
    margin-left: auto;
    margin-right: auto;
    max-width: 200px;
    text-align: center;
    transform: translateY(-50%);
    width: 100%;
}

.counter__number {
    font-size: 2rem;
    font-weight: bold;
    width: 100%;
}

.counter__button {
    flex-grow: 1;
    background-color: #1e88e5;
}

.counter__button:hover, .counter__button:focus {
    background-color: #1976d2;
}