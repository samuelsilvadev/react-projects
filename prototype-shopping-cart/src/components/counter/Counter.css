.counter {
    margin: 1rem 0;
    display: flex;
    justify-content: center;
}

.counter__quantity {
    min-width: 40px;
}

.counter__btn-increment {
	margin-right: 0.5rem;
}
