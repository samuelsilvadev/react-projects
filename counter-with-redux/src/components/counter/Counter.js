import React from 'react';
import { connect } from 'react-redux';

import { INCREMENT, DECREMENT } from './../../state/types';

import './Counter.css';

export const Counter = ({ counter, increment, decrement }) => {
    return (
        <div className="counter card">
            <p data-enzyme-id="counter" className="counter__number">{ counter }</p>
            <button
                data-enzyme-id="counter-decrement-button"
                className="counter__button btn waves-effect waves-light"
                onClick={ decrement }>
                -
            </button>
            <button
                data-enzyme-id="counter-increment-button"
                className="counter__button btn waves-effect waves-light"
                onClick={ increment }>
                +
            </button>
        </div>
    ); 
};

const mapStateToProps = (state) => ({
    counter: state,
});

const mapDispatchToProps = (dispatch) => ({
    increment: () => dispatch({ type: INCREMENT }),
    decrement: () => dispatch({ type: DECREMENT }),
});

export default connect(mapStateToProps, mapDispatchToProps)(Counter);