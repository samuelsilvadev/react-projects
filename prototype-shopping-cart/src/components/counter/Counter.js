import React from 'react';
import PropTypes from 'prop-types';

import './Counter.css';

class Counter extends React.Component {

	render() {
		return (
			<div className="counter">
				<span
					className={this._getBadgeClasses()}>
					{this._formatCount()}
				</span>
				<button
					className="btn btn-primary btn-sm counter__btn-increment"
					onClick={this._handleClickIncrementBtn}>
					Increment
				</button>
				<button
					className="btn btn-danger btn-sm"
					onClick={this._handleClickDeleteBtn}>
					Delete
				</button>
			</div>
		);
	}

	_handleClickIncrementBtn = () => {
		const { onIncrement, id } = this.props;
		onIncrement(id);
	}

	_handleClickDeleteBtn = () => {
		const { onDelete, id } = this.props;
		onDelete(id);
	}

	_formatCount() {
		const { value } = this.props;
		return (value === 0 ? 'Zero' : value);
	}

	_getBadgeClasses() {
		const { value } = this.props;
		return `counter__quantity m-2 badge badge-${value === 0 ? 'warning' : 'success'}`;
	}
}

Counter.propTypes = {
	id: PropTypes.number.isRequired,
	value: PropTypes.number,
	onDelete: PropTypes.func,
	onIncrement: PropTypes.func,
};

Counter.defaultProps = {
	value: 0,
	onDelete: () => { },
	onIncrement: () => { },
};

export default Counter;
