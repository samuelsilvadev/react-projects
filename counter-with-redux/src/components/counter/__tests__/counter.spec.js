import React from 'react';
import { shallow } from 'enzyme';

import { Counter } from './../Counter';

describe('<Counter />', () => {
    describe('Unconnected counter', () => {
        it('should render component correctly without props', () => {
            const tree = shallow(<Counter />);
    
            expect(tree).toMatchSnapshot();
        });
    
        it('should render component correctly with props', () => {
            const props = {
                counter: 1,
                increment: () => {},
                decrement: () => {},
            };
            const tree = shallow(<Counter { ...props } />);
            const $counter = tree.find('[data-enzyme-id="counter"]');
    
            expect($counter.text()).toBe('1');
            expect(tree).toMatchSnapshot();
        });
    
        it('should call `increment` if click on increment button', () => {
            const props = {
                counter: 1,
                increment: jest.fn(),
            };
            const tree = shallow(<Counter { ...props } />);
            const $incrementButton = tree.find('[data-enzyme-id="counter-increment-button"]');
    
            $incrementButton.simulate('click');
    
            expect(props.increment).toHaveBeenCalledTimes(1);
        });
    
        it('should call `decrement` if click on decrement button', () => {
            const props = {
                counter: 1,
                decrement: jest.fn(),
            };
            const tree = shallow(<Counter { ...props } />);
            const $decrementButton = tree.find('[data-enzyme-id="counter-decrement-button"]');
    
            $decrementButton.simulate('click');
    
            expect(props.decrement).toHaveBeenCalledTimes(1);
        });
        
    });
});