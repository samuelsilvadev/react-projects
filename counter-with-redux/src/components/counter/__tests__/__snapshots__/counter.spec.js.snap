// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> Unconnected counter should render component correctly with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter
    counter={1}
    decrement={[Function]}
    increment={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="counter__number"
          data-enzyme-id="counter"
        >
          1
        </p>,
        <button
          className="counter__button btn waves-effect waves-light"
          data-enzyme-id="counter-decrement-button"
          onClick={[Function]}
        >
          -
        </button>,
        <button
          className="counter__button btn waves-effect waves-light"
          data-enzyme-id="counter-increment-button"
          onClick={[Function]}
        >
          +
        </button>,
      ],
      "className": "counter card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1,
          "className": "counter__number",
          "data-enzyme-id": "counter",
        },
        "ref": null,
        "rendered": 1,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "-",
          "className": "counter__button btn waves-effect waves-light",
          "data-enzyme-id": "counter-decrement-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "-",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "className": "counter__button btn waves-effect waves-light",
          "data-enzyme-id": "counter-increment-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="counter__number"
            data-enzyme-id="counter"
          >
            1
          </p>,
          <button
            className="counter__button btn waves-effect waves-light"
            data-enzyme-id="counter-decrement-button"
            onClick={[Function]}
          >
            -
          </button>,
          <button
            className="counter__button btn waves-effect waves-light"
            data-enzyme-id="counter-increment-button"
            onClick={[Function]}
          >
            +
          </button>,
        ],
        "className": "counter card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": "counter__number",
            "data-enzyme-id": "counter",
          },
          "ref": null,
          "rendered": 1,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-",
            "className": "counter__button btn waves-effect waves-light",
            "data-enzyme-id": "counter-decrement-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "-",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "className": "counter__button btn waves-effect waves-light",
            "data-enzyme-id": "counter-increment-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Counter /> Unconnected counter should render component correctly without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="counter__number"
          data-enzyme-id="counter"
        />,
        <button
          className="counter__button btn waves-effect waves-light"
          data-enzyme-id="counter-decrement-button"
        >
          -
        </button>,
        <button
          className="counter__button btn waves-effect waves-light"
          data-enzyme-id="counter-increment-button"
        >
          +
        </button>,
      ],
      "className": "counter card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "counter__number",
          "data-enzyme-id": "counter",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "-",
          "className": "counter__button btn waves-effect waves-light",
          "data-enzyme-id": "counter-decrement-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "-",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "className": "counter__button btn waves-effect waves-light",
          "data-enzyme-id": "counter-increment-button",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="counter__number"
            data-enzyme-id="counter"
          />,
          <button
            className="counter__button btn waves-effect waves-light"
            data-enzyme-id="counter-decrement-button"
          >
            -
          </button>,
          <button
            className="counter__button btn waves-effect waves-light"
            data-enzyme-id="counter-increment-button"
          >
            +
          </button>,
        ],
        "className": "counter card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "counter__number",
            "data-enzyme-id": "counter",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-",
            "className": "counter__button btn waves-effect waves-light",
            "data-enzyme-id": "counter-decrement-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "-",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "className": "counter__button btn waves-effect waves-light",
            "data-enzyme-id": "counter-increment-button",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
